aiida/tools/graph/graph_traversers.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Module for functions to traverse AiiDA graphs."""
+
+from __future__ import absolute_import
+from numpy import inf
+
+
+def traverse_graph(starting_pks, max_iterations=None, get_links=False, **kwargs):
```

Comment:
add argument `graph_traversal_rules` which should be an instance of `aiida.common.links.GraphTraversalRules` and will serve as the validation mapping. I.e. if use passes `GraphTraversalRules.VISUALIZE` those are the rules that will be used for the default as well as to validate the overrides in the `kwargs`. We can at some point make syntactic sugar or wrapping functions to not have to explicitly pass the enum designating the "operation" e.g. export or delete